java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1291,4 +1291,24 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_title(JNIEnv *env, jobj
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_nansToNulls(JNIEnv *env, jobject j_object, jlong handle) {
+
+  JNI_NULL_CHECK(env, handle, "native view handle is null", 0)
+
+  try {
+    cudf::jni::auto_set_device(env);
+    cudf::column_view * view = reinterpret_cast<cudf::column_view *>(handle);
+    // get a new null mask by setting all the nans to null
+    std::pair<std::unique_ptr<rmm::device_buffer>, cudf::size_type> pair = cudf::experimental::nans_to_nulls(*view);
+    // create a column_view which is a no-copy wrapper around the original column without the null mask
+    std::unique_ptr<cudf::column_view> copy_view(new cudf::column_view(view->type(), view->size(), view->data<char>()));
+    // create a column by deep copying the copy_view
+    std::unique_ptr<cudf::column> copy(new cudf::column(*copy_view));
+    // set the null mask with nans set to null
+    copy->set_null_mask(std::move(*pair.first), pair.second);
```

Comment:
Don't we need to call `release` on the `unique_ptr` here to transfer ownership of the `device_buffer`?

Reply:
It's not needed as per the docs https://en.cppreference.com/w/cpp/memory/unique_ptr/operator%3D

Reply:
But we're not using `unique_ptr`'s assignment operator here.  We're invoking `std::move` on the result of `unique_ptr`'s dereference operator, which doesn't inform `unique_ptr` of what's going on.  It will still think it owns the `device_buffer` and destroy it when the `pair` variable falls out of scope at the end of this method.

It think this needs to be something like:
```c++
  copy->set_null_mask(std::move(*(pair.first.release())), pair.second);
```

Reply:
aah. I see. Thanks for pointing this out.